src/blockdata/transaction.rs

```diff
@@ -1126,19 +1127,36 @@ mod tests {
         spent.insert(spent1.txid(), spent1);
         spent.insert(spent2.txid(), spent2);
         spent.insert(spent3.txid(), spent3);
+        let mut spent2 = spent.clone();
+        let mut spent3 = spent.clone();
 
-        let resolver = |point: &OutPoint| {
-            if let Some(tx) = spent.get(&point.txid) {
+        spending.verify(|point: &OutPoint| {
+            if let Some(tx) = spent.remove(&point.txid) {
                 return tx.output.get(point.vout as usize).cloned();
             }
             None
```

Comment:
This and other following tests can be simplified to something like: 
```
spent.remove(point.txid).map(|tx| tx.output.get(point.vout as usize))
```

Reply:
spent.remove(&point.txid).and_then(|tx| tx.output.get(point.vout as usize).cloned()) would work as otherwise it would be Option<Option<TxOut>> also tried to return reference here end got into trouble, not worth it. Since tests are readable I leave as is.